eis_qgis_plugin/processing/algorithms/exploratory_analysis/local_morans_i.py

```diff
@@ -0,0 +1,80 @@
+from qgis.core import (
+    QgsProcessingParameterEnum,
+    QgsProcessingParameterFeatureSource,
+    QgsProcessingParameterField,
+    QgsProcessingParameterNumber,
+    QgsProcessingParameterVectorDestination,
+)
+
+from eis_qgis_plugin.processing.eis_processing_algorithm import EISProcessingAlgorithm
+
+
+class EISLocalMoransI(EISProcessingAlgorithm):
+    def __init__(self) -> None:
+        super().__init__()
+
+        self._name = "local_morans_i"
+        self._display_name = "Local Moran's I"
+        self._group = "Exploratory analysis"
+        self._group_id = "exploratory_analysis"
+        self._short_help_string = (
+            "Perform Local Moran's I calculation."
+        )
+
+    def initAlgorithm(self, config=None):
+        self.alg_parameters = [
+            "input_vector",
+            "output_vector",
+            "column",
+            "weight_type",
+            "k",
+            "permutations",
+        ]
+
+        input_vector_param = QgsProcessingParameterFeatureSource(
+            name=self.alg_parameters[0],
+            description="Input geometries"
+        )
+        input_vector_param.setHelp("Input geometries that contains the data to be examined with Local Moran's I.")
+        self.addParameter(input_vector_param)
+
+        output_vector_param = QgsProcessingParameterVectorDestination(
+            name=self.alg_parameters[1],
+            description="Output geometries"
+        )
+        output_vector_param.setHelp("Output for Local Moran's I calculation.")
```

Comment:
We could describe the output a bit more here since we are adding the help string, something like this:

"Output vector file with attributes for Local Moran's I statistic and p-value for the statistic."

Reply:
Also, you can (and perhaps should for consistency) add the output parameter definition as the last. That way it is easy to find in the code and is in the same place it appears in the UI. It is often directly after input file in the CLI function definitions because parameters with default values must come last, and the file input parameters don't have default values. Because we use (mostly) Typer options, the different ordering of parameters here and in CLI does not brake anything